nixos/modules/services/security/fprintd.nix

```diff
@@ -27,27 +28,56 @@ in
 
       package = mkOption {
         type = types.package;
-        default = pkgs.fprintd;
-        defaultText = "pkgs.fprintd";
+        default = fprintdPkg;
+        defaultText = "pkgs.fprintd and if tod.enable=`true` then pkgs.fprintd-tod";
         description = ''
           fprintd package to use.
         '';
       };
 
+      tod = {
+
+        enable = mkEnableOption "Touch OEM Drivers library support";
+
+        driver =  {
+
+          package = mkOption {
+            type = types.package;
+            default = pkgs.libfprint-2-tod1-goodix;
+            defaultText = "pkgs.libfprint-2-tod1-goodix";
+            description = ''
+              Touch OEM Drivers (TOD) package to use.
+            '';
+          };
+
+          path = mkOption {
```

Comment:
Is there only 1 possible path per package, or can a single package contain multiple of these paths?

Reply:
I have no answer for this, that's why I left the ability to specify the path. It seems for me the only reasonable solution in case someone wants to use their custom pkgs

Reply:
Alright. **If** the answer was that there is only 1 path per package I was going to suggest we do what we do with shell packages... you could add `passthru.driverPath = "/usr/lib/libfprint-2/tod-1"; # or whatever` to each possible driver package and then avoid the `driver.path` option entirely, automatically using `cfg.driver.driverPath` and saving the user having to specify the path. This is how things work when you type `users.users.ymatsiuk.shell = pkgs.bash`.

Reply:
Oh! That makes total sense, let me dig into this a bit more.

Reply:
Awesome. Let me know if you need a hand or whatever.